homeassistant/components/plugwise/sensor.py

```diff
@@ -0,0 +1,434 @@
+"""Plugwise Sensor component for Home Assistant."""
+
+import logging
+
+from homeassistant.const import (
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_ILLUMINANCE,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_PRESSURE,
+    DEVICE_CLASS_TEMPERATURE,
+    ENERGY_KILO_WATT_HOUR,
+    ENERGY_WATT_HOUR,
+    POWER_WATT,
+    PRESSURE_BAR,
+    TEMP_CELSIUS,
+    UNIT_PERCENTAGE,
+    VOLUME_CUBIC_METERS,
+)
+from homeassistant.helpers.entity import Entity
+
+from . import SmileGateway
+from .const import (
+    COOL_ICON,
+    DEVICE_STATE,
+    DOMAIN,
+    FLAME_ICON,
+    IDLE_ICON,
+    SENSOR_MAP_DEVICE_CLASS,
+    SENSOR_MAP_ICON,
+    SENSOR_MAP_MODEL,
+    SENSOR_MAP_UOM,
+    UNIT_LUMEN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_TEMPERATURE = [
+    "Temperature",
+    TEMP_CELSIUS,
+    DEVICE_CLASS_TEMPERATURE,
+    "mdi:thermometer",
```

Comment:
Entities with device class set should not set a custom icon. The icon is already set in the frontend.

Reply:
What should `def icon` return from our class than? (or shouldn't it be defined at all unless an icon is set)?

Reply:
Not defined at all is ok, since there's a default in the base entity class that returns `None`. Otherwise return `None`.

Reply:
Removed accordingly